xbmc/interfaces/AnnouncementManager.cpp

```diff
@@ -248,5 +276,29 @@ void CAnnouncementManager::Announce(AnnouncementFlag flag, const char *sender, c
   if (id > 0)
     object["item"]["id"] = id;
 
-  Announce(flag, sender, message, object);
+  DoAnnounce(flag, sender, message, object);
+}
+
+void CAnnouncementManager::Process()
+{
+  SetPriority(GetMinPriority());
+
+  while (!m_bStop)
+  {
+    CSingleLock lock (m_critSection);
+    if (!m_announcementQueue.empty())
+    {
+      auto announcement = m_announcementQueue.front();
+      m_announcementQueue.pop_front();
+      {
+        CSingleExit ex(m_critSection);
+        DoAnnounce(announcement.flag, announcement.sender.c_str(), announcement.message.c_str(), announcement.item, announcement.data);
+      }
+    }
+    else
+    {
+      CSingleExit ex(m_critSection);
+      m_queueEvent.WaitMSec(1000);
```

Comment:
>I am not in the mood for bikeshedding.

Yeah, thanks, the feeling is mutual. This is basic producer-consumer. You seem to have missed the point: it's not about the work that it's doing but about waking up. We have so many of these in our code base for no reason and it kills performance. Why wake up just to poll the queue for changes when you can sleep instead?

When I see someone resorting to polling it tells me they haven't understood the problem, or that there's a bug that's being worked around by polling it. Just change 1000 to infinite an see. If it works, we're good, otherwise this issue will be clear.